Docs/04-GeneratingCertificates.md

```diff
@@ -0,0 +1,103 @@
+# Generating Certificates
+
+This document will walk you through one way to generate a CA and self-signed certificate for use with Steward.  
+
+Two-way trust must be established between Steward and the Strategy Provider client for each request. To accomplish this, both the Server and Client will need TLS certificates and the CA certificate of the other party (Steward needs the client's CA, and the client needs Steward's CA). To make generating Steward's certificates easier, we've provided a convenience script called `gen-script.sh` which can be found [here](https://github.com/peggyjv/steward/gen-script.sh). The script takes one optional flag for the output location of the certificates and their signing keys. 
+
+## Steps
+
+0. Ensure you have OpenSSL installed.
+
+This script requires OpenSSL to be installed; it will not work with LibreSSL.
+
+Note that LibreSSL still uses the binary name `openssl`. You'll know if you have the wrong binary if `openssl version` says "LibreSSL".
+
+1. Run the following command from the root of this repository
+
+```
+./gen-certs.sh -o some/output/location/
+```
+
+If you don't set the output path flag, certificates and keys will be created in the current working directory.
+
+2. Fill out the Server CA information
+
+After running this command you run through a series of prompts. The first will be for information pertaining to the certificate authority like geographical location, email, etc. These fields can be safely left blank. 
+
+3. Fill out the Server Certificate information
+
+A similar set of prompts will ask for the same information for the Server (Steward) certificate which will be signed by the CA. These fields can also be left blank, though you may want to set the Common Name to your domain name.
+
+4. Provide your IP and/or domain name
+
+The final two prompts will ask for the IP and/or domain name where Steward will be publically hosted. At least one of these MUST be set for Steward to start up with the certificate.
```

Comment:
I think since the pubsub future will only allow domain names, let's just drop IP support and ask them to use a domain/subdomain.